internal/cli/run/run.go

```diff
@@ -50,20 +48,20 @@ func init() {
 	cmd := root.Command("run", "Run a test group or OONI Run link")
 
 	var nettestGroupNamesBlue []string
+	var ctx *ooni.Context
+	var network *database.Network
+
 	for name := range groups.NettestGroups {
 		nettestGroupNamesBlue = append(nettestGroupNamesBlue, color.BlueString(name))
 	}
 
-	nettestGroup := cmd.Arg("name",
-		fmt.Sprintf("the nettest group to run. Supported tests are: %s, or nothing to run them all",
-			strings.Join(nettestGroupNamesBlue, ", "))).String()
-
 	noCollector := cmd.Flag("no-collector", "Disable uploading measurements to a collector").Bool()
 	collectorURL := cmd.Flag("collector-url", "Specify the address of a custom collector").String()
 	bouncerURL := cmd.Flag("bouncer-url", "Specify the address of a custom bouncer").String()
 
-	cmd.Action(func(_ *kingpin.ParseContext) error {
-		ctx, err := root.Init()
+	cmd.Action(func(kp *kingpin.ParseContext) error {
+		var err error
+		ctx, err = root.Init()
```

Comment:
I have a more scratching-my-head-like question: why is `ctx` initialized here and used in later commands?

Reply:
This is the parent action, so it's going to be called anytime you call the child actions. The initialisation of ctx is done in here so that we don't have to duplicate the code for every sub-action.

Reply:
While it is true that we don't have to duplicate the code, a much less confusing implementation would be to factor the initialisation code in a function that returns whatever we need, and call it. The current implementation is magic and, because of this reason, I don't like that [*]. I would be happier if we could have more plain code here. Can you please refactor the initialisation code and have it return what you need in every context?

[*] This is, in a way, similar to using a global variable, except that a programmer does not expect this to happen because they are usually more accustomed to global variables being dangerous, not to stack variables. My issue with this usage of a stack variable is that it confuses my brain when I reason about scoping.

Reply:
This was also done in the interest of making changes part of this PR minimal. Had I done what you suggest I probably would have had to make more substantial changes in how subcommands are handled.

I agree that this can maybe be improved. Would you like to maybe document this as an issue and work on refactoring this code?

Reply:
This change is adding technical debt. Can we avoid that now rather than postponing?

Reply:
Sure, feel free to push code to this branch as well.

Reply:
I migrated this to an issue: https://github.com/ooni/probe-cli/issues/55

Reply:
I have migrated this to issue #55